app/assets/javascripts/sitewide/dropdown_selector.js

```diff
@@ -0,0 +1,455 @@
+            updateTags(selector, container);
+          }
+        }, 300);
+      });
+    }
+
+    container.find('.ds-tags').remove();
+    if (selector.data('combine-tags')) {
+      if (selectedOptions.length === 1) {
+        drawTag({ label: selectedOptions[0].label, value: selectedOptions[0].value });
+      } else if (allOptionsSelected(selector, container)) {
+        drawTag({ label: selector.data('select-multiple-all-selected'), value: 0 });
+      } else if (selectedOptions.length > 1) {
+        drawTag({ label: `${selectedOptions.length} ${selector.data('select-multiple-name')}`, value: 0 });
+      }
+    } else {
+      $.each(selectedOptions, (ti, tag) => {
+        drawTag(tag);
+      });
+    }
+
+    searchFieldValue.attr('placeholder',
+      selectedOptions.length > 0 ? '' : selector.data('placeholder'));
+
+    if (!selector.data('combine-tags')) {
+      container.find('.ds-tags').addClass('stretch');
+    } else {
+      container.find('.ds-tags').removeClass('stretch');
+    }
+
+    updateDropdownDirection(selector, container);
+    refreshDropdownSelection(selector, container);
+    if (selector.data('config').onChange && !config.skipChange) {
+      selector.data('config').onChange();
+    }
+  }
+
+  // Convert local data or ajax data to same format
+  function dataSource(selector, container) {
+    var result = [];
+    var groups;
+    var options;
+    var defaultParams;
+    var customParams;
+    var ajaxParams;
+    if (selector.data('ajax-url')) {
+      defaultParams = { query: container.find('.seacrh-field').val() };
+      customParams = selector.data('config').ajaxParams;
+      ajaxParams = customParams ? customParams(defaultParams) : defaultParams;
+
+      $.get(selector.data('ajax-url'), ajaxParams, (data) => {
+        loadData(selector, container, data);
+      });
+    } else if (selector.data('select-by-group')) {
+      groups = selector.find('optgroup');
+      $.each(groups, (gi, group) => {
+        var groupElement = { label: group.label, value: group.label, options: [] };
+        var groupOptions = filterOptions(selector, container, $(group).find('option'));
+        $.each(groupOptions, function(oi, option) {
+          groupElement.options.push({ label: option.innerHTML, value: option.value });
+        });
+        if (groupElement.options.length > 0) result.push(groupElement);
+      });
+    } else {
+      options = filterOptions(selector, container, selector.find('option'));
+      $.each(options, function(oi, option) {
+        result.push({ label: option.innerHTML, value: option.value });
+      });
+    }
+    return result;
+  }
+
+  // ////////////////////
+  // Public functions ///
+  // ////////////////////
+
+  return {
+    // Dropdown initialization
+    init: function(selector, config) {
+      generateDropdown(selector, config);
+
+      return this;
+    },
+
+    // Clear button initialization
+    initClearButton: function(selector, clearButton) {
```

Comment:
"clear" button it is optional for initialization of dropdown. Because it is not part of dropdown. And by default it must not have clear button. For example experiment and task don't have there own clear button, because they connected to project.